include/pistache/mailbox.h

```diff
@@ -155,6 +155,10 @@ class Queue {
     struct Entry {
         friend class Queue;
 
+        Entry () {
+            memset(&storage, 0, sizeof(storage));
```

Comment:
Will this template ever be instantiated with a non-POD data type?
Consider instead:

   Entry(): storage() {}

Reply:
I'm not sure.  To be honest it's possible this PR isn't even relevant anymore, it's been almost a year.  I think you make a good point though.